### PR TITLE
Simplified the IRulesEngine interface to have a params array for Obje…

### PR DESCRIPTION
**Pull request by toddmeinershagen**
…cts and Rule Parameters.  Also updated the code to use the ILogger from Microsoft.Extensions.Logging.

### CODE REVIEW

File: src/RulesEngine/RulesEngine/NullLogger.cs [hunk 1]
```diff
@@ -1,18 +1,27 @@
-﻿// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
-
 using System;
```
Comment (abbasc52) [inline src/RulesEngine/RulesEngine/NullLogger.cs:1]: Can we remove this as well and replace with - https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.abstractions.nulllogger?view=dotnet-plat-ext-3.1 ?

Comment (toddmeinershagen) [reply src/RulesEngine/RulesEngine/NullLogger.cs:1]: Yes - I like that as well.

Comment (toddmeinershagen) [reply src/RulesEngine/RulesEngine/NullLogger.cs]: @abbasc52 I have made the updates using ```NullLogger<RulesEngine>``` instead.  Let me know if you find anything else and when/if you can release a new version.

Comment (abbasc52) [reply src/RulesEngine/RulesEngine/NullLogger.cs]: Thanks. We are actually looking at redefining our namespace so we will be taking that up and release version 2 of the nuget. We are planning to release the new version by next week.

**Pull request merged**